hack/deployer/runner/ocp3.go

```diff
@@ -0,0 +1,196 @@
+		return err
+	}
+
+	switch d.plan.Operation {
+	case DeleteAction:
+		if err := d.runAnsibleDockerContainer("delete"); err != nil {
+			return err
+		}
+	case CreateAction:
+		if err := d.runAnsibleDockerContainer("create"); err != nil {
+			return err
+		}
+
+		if err := d.GetCredentials(); err != nil {
+			return err
+		}
+
+		if err := setupDisks(d.plan); err != nil {
+			return err
+		}
+
+		if err := createStorageClass(); err != nil {
+			return err
+		}
+	default:
+		err = fmt.Errorf("unknown operation %s", d.plan.Operation)
+	}
+
+	return err
+}
+
+func writeGCloudSSHKey(vaultInfo VaultInfo) error {
+	log.Printf("Setting GCP SSH keys...")
+	sshDir := filepath.Join(os.Getenv("HOME"), ".ssh")
+	_ = os.MkdirAll(sshDir, os.ModePerm)
+
+	client, err := NewClient(vaultInfo)
+	if err != nil {
+		return err
+	}
+
+	keyFileName := filepath.Join(sshDir, "google_compute_engine")
+	if err := client.ReadIntoFile(keyFileName, OcpVaultPath, Ocp3GCloudPrivateSSHKeyFieldName); err != nil {
+		return err
+	}
+	pubKeyFileName := filepath.Join(sshDir, "google_compute_engine.pub")
+	if err := client.ReadIntoFile(pubKeyFileName, OcpVaultPath, Ocp3GCloudPublicSSHKeyFieldName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d Ocp3Driver) writeAnsibleVarsFile() error {
+	log.Printf("Setting Ansible vars...")
+
+	vars := struct {
+		MasterCount    int      `yaml:"master_count"`
+		MasterInstance string   `yaml:"master_instance"`
+		WorkerCount    int      `yaml:"worker_count"`
+		WorkerInstance string   `yaml:"worker_instance"`
+		AllowedSources []string `yaml:"allowed_sources"`
+	}{
+		MasterCount:    MasterCount,
+		MasterInstance: MasterInstance,
+		WorkerCount:    d.plan.Ocp3.WorkerCount,
+		WorkerInstance: d.plan.MachineType,
+		AllowedSources: []string{"0.0.0.0/0"}, // TODO: get current IP to be more restrictive?
+	}
+	varsBytes, err := yaml.Marshal(vars)
+	if err != nil {
+		return err
+	}
+	varsFile := filepath.Join(os.Getenv("HOME"), AnsibleVarsFilename)
+	if err := ioutil.WriteFile(varsFile, varsBytes, 0644); err != nil {
+		return fmt.Errorf("while writing Ansible variables file %w", err)
+	}
+	return nil
+}
+
+func (d Ocp3Driver) runAnsibleDockerContainer(action string) error {
+	log.Printf("Creating OCP3 cluster with Ansible in Docker...")
+
+	params := map[string]interface{}{
+		"ClusterName":         d.plan.ClusterName,
+		"HomeVolumeName":      SharedVolumeName,
+		"HomeVolumeMountPath": AnsibleHomePath,
+		"GCloudCredsPath":     filepath.Join(AnsibleHomePath, GCPDir, ServiceAccountFilename),
+		"GCloudSDKPath":       filepath.Join(AnsibleHomePath, GCPDir),
+		"AnsibleVarsPath":     filepath.Join(AnsibleHomePath, AnsibleVarsFilename),
+		"AnsibleOutputPath":   filepath.Join(AnsibleHomePath, AnsibleOutputDirname),
+		"AnsibleAction":       action,
+		"AnsibleDockerImage":  AnsibleDockerImage,
+	}
+
+	return NewCommand(`docker run --rm \
+		-e FORCED_GROUP_ID=1000 \
+		-e FORCED_USER_ID=1000 \
+		-e USER={{.ClusterName}} \
```

Comment:
Unfortunately using a different user for each test will create a new entry in the SSH keys of the project:

![image](https://user-images.githubusercontent.com/976373/94141373-eb8c8780-fe6c-11ea-9201-21ec2908042b.png)

I think we have to use a fixed username.

Reply:
Well seen. Anyway, it was not very clean to rely on the current user to name the gcp resources.
I hesitate to introduce a variable `base_name` and replace `user` in the name of the gcp resources that Ansible playbooks are independent of the user name,
or just set the variable `suffix` which is almost used to name all the gcp resources. I think it is not used only for `k8s-{{ user }}-vpc`, `k8s-{{ user }}-firewall` and `k8s-{{ user }}-firewall-internal`.

Reply:
> I hesitate to introduce a variable base_name and replace user in the name of the gcp resources that Ansible playbooks are independent of the user name
 
I used to rely on the user name because it was convenient for humans.

> just set the variable `suffix` which is almost used to name all the gcp resources

If you mean having a fixed `user` (e.g. `jenkins`) but a dynamic `suffix`, I think it should work without having to update the image. I would start here.

Reply:
6d4483d adresses that.